packages/transaction_signing_gateway/lib/model/transaction_response.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:alan/alan.dart';
+import 'package:decimal/decimal.dart';
+import 'package:equatable/equatable.dart';
+
+class TransactionResponse extends Equatable {
+  const TransactionResponse({
+    required this.hash,
+    required this.data,
+    required this.logs,
+    required this.info,
+    required this.codespace,
+    required this.rawLog,
+    required this.gasUsed,
+    required this.gasWanted,
+    required this.timestamp,
+  });
+
+  factory TransactionResponse.fromTxResponse(TxResponse tx) {
```

Comment:
let's try to always keep the models agnostic of the underlying libraries, thus the factory that we have here we could add a separate file `model_translations` and add this as an extension:

```dart
extension TxResponseTranslator on TxResponse {
TransactionResponse toTransactionResponse() => TransactionResponse( ... );
```
this is because we might want to support other chains later in the future (like Ethereum) and we want to adhere to the [Open-Closed Principle](https://deviq.com/principles/open-closed-principle)

Reply:
Added fixes for this